frontends/common/constantFolding.cpp

```diff
@@ -275,6 +279,32 @@ DoConstantFolding::compare(const IR::Operation_Binary* e) {
             bool bresult = (le->name == re->name) == eqTest;
             return new IR::BoolLiteral(e->srcInfo, bresult);
         }
+
+        auto llist = eleft->to<IR::ListExpression>();
+        auto rlist = eright->to<IR::ListExpression>();
+        if (llist != nullptr && rlist != nullptr) {
+            if (llist->components.size() != rlist->components.size()) {
+                ::error("%1%: comparing lists of different size", e);
+                return e;
+            }
+
+            for (size_t i = 0; i < llist->components.size(); i++) {
+                auto li = llist->components.at(i);
+                auto ri = rlist->components.at(i);
+                const IR::Operation_Binary* tmp;
+                if (eqTest)
+                    tmp = new IR::Equ(li, ri);
+                else
+                    tmp = new IR::Neq(li, ri);
+                auto cmp = compare(tmp);
+                auto boolLit = cmp->to<IR::BoolLiteral>();
+                if (boolLit == nullptr)
+                    return e;
+                if (!boolLit->value)
+                    return boolLit;
```

Comment:
This seems correct for `==` (false if any component differs) but seems wrong for `!=` (should be false only if all components are the same).  Should be `if (boolLit->value != eqTest)`?